qiskit_ibm_runtime/_cli.py

```diff
@@ -0,0 +1,234 @@
+        )
+        while True:
+            token = getpass("Token: ").strip()
+            if token != "":
+                return token
+
+    @classmethod
+    def get_instance(cls, service: QiskitRuntimeService) -> str:
+        """
+        Ask user which instance to use, or select automatically if only one
+        is available.
+        """
+        instances = service.instances()
+        if len(instances) == 1:
+            instance = instances[0]
+            print(f"Using instance {Format.greenbold(instance)}")
+            return instance
+        print(Format.bold("\nSelect a default instance"))
+        return select_from_list(instances)
+
+    @classmethod
+    def save_to_disk(cls, account: dict) -> None:
+        """
+        Save account details to disk, confirming if they'd like to overwrite if
+        one exists already. Display a warning that token is stored in plain
+        text.
+        """
+        try:
+            AccountManager.save(**account)
+        except AccountAlreadyExistsError:
+            response = user_input(
+                message="\nDefault account already exists, would you like to overwrite it? (y/N):",
+                is_valid=lambda response: response.strip().lower() in ["y", "yes", "n", "no", ""],
+            )
+            if response.lower() in ["y", "yes"]:
+                AccountManager.save(**account, overwrite=True)
+            else:
+                print("Account not saved.")
+                return
+
+        print(f"Account saved to {Format.greenbold(_DEFAULT_ACCOUNT_CONFIG_JSON_FILE)}")
+        cls.print_box(
+            [
+                "⚠️ Warning: your token is saved to disk in plain text.",
+                "If on a shared computer, make sure to revoke your token",
+                "by regenerating it in your account settings when finished.",
+            ]
+        )
+
+
+def user_input(message: str, is_valid: Callable[[str], bool]) -> str:
+    """
+    Repeatedly ask user for input until they give us something that satisifies
+    `is_valid`.
+    """
+    while True:
+        response = input(message + " ").strip()
+        if response == "quit":
+            sys.exit()
+        if is_valid(response):
+            return response
+        print("Did not understand input, trying again... (type 'quit' to quit)")
+
+
+def select_from_list(options: List[T]) -> T:
+    """
+    Prompt user to select from a list of options by entering a number.
+    """
+    print()
+    for index, option in enumerate(options):
+        print(f"  ({index+1}) {option}")
+    print()
+    response = user_input(
+        message=f"Enter a number 1-{len(options)} and press enter:",
+        is_valid=lambda response: response.isdigit()
+        and int(response) in range(1, len(options) + 1),
+    )
+    choice = options[int(response) - 1]
+    print(f"Selected {Format.greenbold(str(choice))}")
+    return choice
+
+
+class Format:
```

Comment:
Nice! It's not always safe to use color codes, such as when piping output or in certain terminals. I recommend renaming this to be `Formatter` and setting an attribute `supports_ansi_codes`. Then, change the methods to be instance methods that use `supports_ansi_codes`.

Determining the color code support is tricky. https://stackoverflow.com/questions/7445658/how-to-detect-if-the-console-does-support-ansi-escape-codes-in-python links to Django's approach in the first answer (click the link for the latest). Imo it's not critical that we always correctly detect things; it can be best effort.

It's also common for CLIs to have an option `--colors` and `--no-colors` to force the CLI to behave a certain way.

Reply:
Very sensible. Thanks!